logstash-core/lib/logstash/config/source/local.rb

```diff
@@ -0,0 +1,203 @@
+
+          if temporary_file?(file)
+            logger.warn("NOT reading config file because it is a temp file", :config_file => file)
+            next
+          end
+
+          config_string = ::File.read(file)
+
+          if valid_encoding?(config_string)
+            config_parts << ConfigPart.new(self.class.name, file, config_string)
+          else
+            encoding_issue_files << file
+          end
+        end
+
+        if encoding_issue_files.any?
+          raise LogStash::ConfigLoadingError, "The following config files contains non-ascii characters but are not UTF-8 encoded #{encoding_issue_files}"
+        end
+
+        raise LogStash::ConfigLoadingError, "Cannot load configuration for path: #{path}" if config_parts.empty?
+        config_parts
+      end
+
+      def self.read(path)
+        ConfigPathLoader.new(path).read
+      end
+
+      private
+      def normalize_path(path)
+        path.gsub!(LOCAL_FILE_URI, "")
+        ::File.expand_path(path)
+      end
+
+      def get_files
+        Dir.glob(path).sort
+      end
+
+      def path
+        if ::File.directory?(@path)
+          ::File.join(@path, "*")
+        else
+          @path
+        end
+      end
+
+      def valid_encoding?(content)
+        content.ascii_only? && content.valid_encoding?
+      end
+
+      def temporary_file?(filepath)
+        filepath.match(TEMPORARY_FILE_RE)
+      end
+    end
+
+    class ConfigRemoteLoader
+      def self.read(uri)
+        uri = URI.parse(uri)
+
+        Net::HTTP.start(uri.host, uri.port, :use_ssl => uri.scheme == "https") do |http|
+          request = Net::HTTP::Get.new(uri.path)
+          response = http.request(request)
+
+          # since we have fetching config we wont follow any redirection.
+          case response.code.to_i
+          when 200
+            [ConfigPart.new(self.name, uri.to_s, response.body)]
+          when 302
+            raise LogStash::ConfigLoadingError, I18n.t("logstash.runner.configuration.fetch-failed", :path => uri.to_s, :message => "We don't follow redirection for remote configuration")
+          when 404
+            raise LogStash::ConfigLoadingError, I18n.t("logstash.runner.configuration.fetch-failed", :path => uri.to_s, :message => "File not found")
+          when 403
+            raise LogStash::ConfigLoadingError, I18n.t("logstash.runner.configuration.fetch-failed", :path => uri.to_s, :message => "Permission denied")
+          when 500
+            raise LogStash::ConfigLoadingError, I18n.t("logstash.runner.configuration.fetch-failed", :path => uri.to_s, :message => "500 error on remote host")
+          else
+            raise LogStash::ConfigLoadingError, I18n.t("logstash.runner.configuration.fetch-failed", :path => uri.to_s, :message => "code: #{response.code}, message: #{response.class.to_s}")
+          end
+        end
+      end
+    end
+
+    PIPELINE_ID = LogStash::SETTINGS.get("pipeline.id").to_sym
```

Comment:
since this is evaluated at file loading time, it won't capture changes to `pipeline.id` coming from the yml file, so it's forever `:main`